### PR TITLE
Add Cluster evaluator

### PR DESCRIPTION
**Pull request by Ivanidzo4ka**
Address #312

### CODE REVIEW

File: src/Microsoft.ML/Models/ClusterMetrics.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.Models
+{
+    /// <summary>
+    /// This class contains the overall metrics computed by regression evaluators.
```
Comment (zeahmed) [inline src/Microsoft.ML/Models/ClusterMetrics.cs]: `regression evaluators`-> `cluster evaluators`

File: src/Microsoft.ML/Models/ClusterMetrics.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        /// <remarks>
+        /// DBI is a measure of the how much scatter is in the cluster and the cluster separation.
+        /// </remarks>
+        public double Dbi { get; private set; }
+
+        /// <summary>
+        /// Normalized Mutual Information
+        /// </summary>
+        /// <remarks>
+        /// NMI is a measure of the mutual dependence of the variables.
```
Comment (zeahmed) [inline src/Microsoft.ML/Models/ClusterMetrics.cs]: Can we rephrase it like this.
```/// NMI is a measure of the mutual dependence between the true and predicted cluster labels for instances in the dataset.
/// NMI ranges between 0 and 1 where "0" indicates clustering is random and "1" indicates clustering is perfect w.r.t true labels. 
```

File: src/Microsoft.ML/Models/ClusterMetrics.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        /// Normalized variants work on data that already has cluster labels.
+        /// It returns values from 0 to 1, where higher numbers are better.
+        /// </remarks>
+        public double Nmi { get; private set; }
+
+        /// <summary>
+        /// Average minimum score.
+        /// </summary>
+        /// <remarks>
+        /// This makes sense for K-Means algorithm, where the 'score' is the distance from the centroid to the example.
```
Comment (zeahmed) [inline src/Microsoft.ML/Models/ClusterMetrics.cs]: Can we rephrase the definition to
```
/// AvgMinScore is the average squared-distance of examples from the respective cluster centroids.
/// It is defined as 
/// AvgMinScore  = (1/m) * sum ((xi - c(xi))^2)
/// where m is the number of instances in the dataset.
/// xi is the ith instance and c(xi) is the centriod of the predicted cluster for xi.
```

Comment (Ivanidzo4ka) [reply src/Microsoft.ML/Models/ClusterMetrics.cs]: Thank you Zeeshan!

---
In reply to: [193260519](https://github.com/dotnet/machinelearning/pull/316#discussion_r193260519) [](ancestors = 193260519)

File: src/Microsoft.ML/Models/TrainTestEvaluator.cs [hunk 4]
```diff
@@ -140,6 +140,12 @@ public sealed partial class TrainTestEvaluator
                         environment,
                         experiment.GetOutput(crossValidateOutput.OverallMetrics)).FirstOrDefault();
                 }
+                else if (Kind==MacroUtilsTrainerKinds.SignatureClusteringTrainer)
```
Comment (shauheen) [inline src/Microsoft.ML/Models/TrainTestEvaluator.cs]: This may need a formatting?

**Pull request merged**